### DISCUSSION TITLE
Trigger validation by email

### DISCUSSION THREAD

**Discussion by AlbertoMoricca** [codeigniter4/shield#537] [Q&A] [answered]
Hi,

I read the documentation available but I didn't understand how to trigger the validation of the user by email included in Shield after the user registration.

I edited the Auth.php with the following code as specified in documentation:
`public array $actions = [
        'register' => 'CodeIgniter\Shield\Authentication\Actions\EmailActivator',
        'login'    => null,
    ];`

But my question is: How can I trigger the email activator in the registration controller? Following the last instructions of my registration controller code:

```php
 $users = model('UserModel');
 $user = new User([
                'username'    => $this->request->getVar('email'),
                'email'    => $this->request->getVar('email'),
                'password' => $this->request->getVar('password')
            ]);
 $users->save($user);

 return redirect()->to('/home');
```

Thanks and best regards,
Alberto

**Comment by kenjis** [comment, marked as answer]
See the Shield `RegisterController`:
https://github.com/codeigniter4/shield/blob/develop/src/Controllers/RegisterController.php#L56

**Reply by AlbertoMoricca** [reply]
Thanks for the help. Just another question: Are these controllers active after the Shield framework installation or must be copied in the project to be used (they are only "examples")?

I think the second answer because I copied the code from RegisterController and now say me "file not found" error for the path "auth/a/show".

**Reply by kenjis** [reply]
They are active, not examples. They are used by default.

**Reply by AlbertoMoricca** [reply]
Ok, thanks for the clarification. I will check how to trigger these included components.
